src/Mongolid/DataMapper/DataMapper.php

```diff
@@ -0,0 +1,287 @@
+<?php
+namespace Mongolid\DataMapper;
+
+use Mongolid\Schema;
+use Mongolid\Container\Ioc;
+use Mongolid\Connection\Pool;
+use Mongolid\Cursor\Cursor;
+use MongoDB\Collection;
+use MongoDB\BSON\ObjectID;
+
+/**
+ * The DataMapper class will abstract how an Entity is persisted and retrieved
+ * from the database.
+ * The DataMapper will always use a Schema trought the SchemaMapper to parse the
+ * document in and out of the database.
+ *
+ * @package  Mongolid
+ */
+class DataMapper
+{
+    /**
+     * Name of the schema class to be used
+     * @var string
+     */
+    public $schemaClass = Schema::class;
+
+    /**
+     * Schema object. Will be set after the $schemaClass
+     * @var Schema
+     */
+    public $schema;
+
+    /**
+     * Connections that are going to be used to interact with the database
+     * @var Pool
+     */
+    protected $connPool;
+
+    /**
+     * @param Pool $connPool The connections that are going to be used to interact with the database.
+     */
+    public function __construct(Pool $connPool)
+    {
+        $this->connPool = $connPool;
+    }
+
+    /**
+     * Upserts the given object into database. Returns success if write concern
+     * is acknowledged.
+     *
+     * @param  mixed $object The object used in the operation.
+     *
+     * @return boolean Success (but always false if write concern is Unacknowledged)
```

Comment:
if you have an object with `writeConcern` 1 and logic like that `if($object->save()) { do it; }` that works and you realize that should be `writeConcern 0`, you logic will be broken with that because will return `false`. For me I believe if writeConcern `0` should return `true`to no break it.